gridlod/lod.py

```diff
@@ -367,6 +400,69 @@ def computeErrorIndicatorCoarseFromCoefficients(patch, muTPrime, aPatchOld, aPat
 
     return computeErrorIndicatorCoarseFromGreeks(patch, muTPrime, (deltaMaxTPrime, kappaMaxT))
 
+def computeEftErrorIndicatorCoarse(patch, cetaTPrime, etaT, aPatchOld, aPatchNew, fElementOld, fElementNew):
+    while callable(aPatchOld):
+        aPatchOld = aPatchOld()
+
+    while callable(aPatchNew):
+        aPatchNew = aPatchNew()
+
+    aOld = aPatchOld
+    aNew = aPatchNew
+
+    world = patch.world
+    NPatchCoarse = patch.NPatchCoarse
+    NCoarseElement = world.NCoarseElement
+    NPatchFine = NPatchCoarse*NCoarseElement
+    iElementPatchCoarse = patch.iElementPatchCoarse
+
+    ### In case aNew and aOld dimensions do not match ###
+    if aNew.ndim == 3 and aOld.ndim ==1:
+        aEye = np.tile(np.eye(2), [np.prod(NPatchFine), 1, 1])
+        aOld = np.einsum('tji, t-> tji', aEye, aOld)
+    if aNew.ndim == 1 and aOld.ndim ==3:
+        aEye = np.tile(np.eye(2), [np.prod(NPatchFine), 1, 1])
+        aNew = np.einsum('tji, t -> tji', aEye, aNew)
+
+
+    elementCoarseIndex = util.convertpCoordIndexToLinearIndex(NPatchCoarse-1, iElementPatchCoarse)
+
+    TPrimeFinetStartIndices = util.pIndexMap(NPatchCoarse-1, NPatchFine-1, NCoarseElement)
+    TPrimeFinetIndexMap = util.lowerLeftpIndexMap(NCoarseElement-1, NPatchFine-1)
+
+    TPrimeIndices = np.add.outer(TPrimeFinetStartIndices, TPrimeFinetIndexMap)
+    aTPrime = aNew[TPrimeIndices]
+    aOldTPrime = aOld[TPrimeIndices]
+
+    if aNew.ndim == 3:
+        aInvTPrime = np.linalg.inv(aTPrime)
+        aOldInvTPrime = np.linalg.inv(aOldTPrime)
+        aDiffTPrime = aTPrime - aOldTPrime
+        deltaMaxTPrime = np.sqrt(np.max(np.linalg.norm(np.einsum('Ttij, Ttjk, Ttkl, Ttlm -> Ttim',
+                                                                 aInvTPrime, aDiffTPrime, aDiffTPrime, aOldInvTPrime),
+                                                       axis=(2,3), ord=2),
+                                        axis=1))
+        kappaMaxT = np.sqrt(np.max(np.linalg.norm(np.einsum('tij, tjk -> tik',
+                                                            aOldTPrime[elementCoarseIndex], aInvTPrime[elementCoarseIndex]),
+                                                  axis=(1,2), ord=2)))
+    else:
+        deltaMaxTPrime = np.max(np.abs((aTPrime - aOldTPrime)/np.sqrt(aTPrime*aOldTPrime)), axis=1)
+        kappaMaxT = np.sqrt(np.max(np.abs(aOldTPrime[elementCoarseIndex] / aTPrime[elementCoarseIndex])))
+
+    # #TODO: THIS IS WRONG... I tried to compute L2 norm here
+
+    # P = fem.assembleProlongationMatrix(patch.NPatchCoarse, patch.world.NCoarseElement)
+    # MLoc = fem.assemblePatchMatrix(patch.NPatchFine, world.MLocFine)
```

Comment:
This should work, I think:
```suggestion
    MElement = fem.assemblePatchMatrix(patch.NCoarseElement, world.MLocFine)
    nuT = np.dot((fElementOld - fElementNew), MElement * (fElementOld - fElementNew))
```
and similarly for gamma. Now, this is provided fElementOld contains the values of f inside the element T, but the name clearly suggests so :)

Reply:
Actually, the name suggest so but that's not the case right now. I just took the node values and I called it 'Element' because it is not the full f but only the nodes for one element.